tests/conftest.py

```diff
@@ -237,3 +236,28 @@ def _keyfunc(obj):
         return cls.__module__, cls.__name__, obj.name
 
     return sorted(iterable, key=_keyfunc)
+
+
+@pytest.fixture
+def noguimods():
```

Comment:
so this is a bit awkward, since now two hard-coded lists of ignored modules exist in the code and here in the fixture. The `ImportBlacklist` class could potentially come in handy for other cases in the future, and I, in general really like to have this mechanism.
I don't think it it's easily possible to ~~hack~~ adapt the code, such that one would only have a single point of definition of blacklisted modules (especially given that they are only blacklisted on certain execution conditions...)

Reply:
Now when I look at this PR I have to admit it looks over-engineered to me. Aside from testing, I also don't see any viable use cases for changing import blacklists more than once per process run. In other words, it seems cleaner to blacklist the needed modules just once as early as possible, and also don't test that at all, because additional loading/unloading complexity required for these tests will be used only for testing.

To summarize, I think we should leave just `ImportBlacklist.find_spec` (because it is needed for the blacklisting to work) and `if args.headless:blacklist_modules("PyQt")`, while removing context managers, exceptions and tests.

Reply:
Alternatively, one can still test this thing by creating a tiny test script that blacklists something, and then running it as a separate process:

```python
def test_module_blacklist():
    assert subprocess.run(["python", "script_that_blacklists_modules.py"]).returncode == 0
```

```python
# script_that_blacklists_modules.py
from ilastik.utility import imp
imp.ModuleBlacklist("spam")
import spam
```